File: 1309-decrypt-string-from-alphabet-to-integer-mapping/1309-decrypt-string-from-alphabet-to-integer-mapping.java
class Solution {
    public String freqAlphabets(String s) {
        StringBuilder sb = new StringBuilder();
        for(int i=s.length()-1;i>=0;i--){
            int number;
            if(s.charAt(i) == '#'){
                number = (s.charAt(i-2) - '0') * 10 + (s.charAt(i-1) - '0');
                i-=2;
            }else{
                number = s.charAt(i) -'0';
            }
            sb.append((char)(number+96));
        }
        return sb.reverse().toString();
    }
}

File: 1309-decrypt-string-from-alphabet-to-integer-mapping/1309-decrypt-string-from-alphabet-to-integer-mapping.js
/**
 * @param {string} s
 * @return {string}
 */
var freqAlphabets = function(s) {
    let str = "";
    for(let i=s.length-1;i>=0;i--){
        let number;
        if(s.charAt(i) === '#'){
            number = (+s.charAt(i-2) * 10) + (+s.charAt(i-1));
            i-=2;
        }else{
            number = +s.charAt(i);
        }
        str += String.fromCharCode(number + 96);
    }
    return str.split('').reverse().join('');
};